ops/model.py

```diff
@@ -3792,6 +3792,40 @@ def _ensure_loaded(self):
         assert self._notice.key == self.key
 
 
+class LazyCheckInfo:
+    """Provide lazily-loaded access to a Pebble check's info.
+
+    The attributes provided by this class are the same as those of
+    :class:`ops.pebble.CheckInfo`, however, the notice details are only fetched
+    from Pebble if necessary (and cached on the instance).
+    """
+
+    name: str
+    level: Optional[Union[pebble.CheckLevel, str]]
+    status: Union[pebble.CheckStatus, str]
+    failures: int
+    threshold: int
+    change_id: Optional[pebble.ChangeID]
+
+    def __init__(self, container: Container, name: str):
+        self._container = container
+        self.name = name
+        self._info: Optional[ops.pebble.CheckInfo] = None
+
+    def __repr__(self):
+        return f'LazyCheckInfo(name={self.name!r})'
+
+    def __getattr__(self, item: str):
+        # Note: not called for defined attribute `name`.
```

Comment:
Sorry, not sure what you mean by this. Do you mean the `item` argument for `__getattr__`? That's a regular Python magic method, so it has the normal meaning, it's the (otherwise undefined) attribute that someone is trying to read.

Reply:
Oh, I didn't even think of that, I guess the comment was redundant for me.
Maybe it's useful for someone else who reads this code some day, so I'm fine with the comment.